### PR TITLE
fix: add user-id header when call gittar api

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What type of this PR
/kind feature

#### What this PR does / why we need it:
fix: add user-id header when call gittar api

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      add user-id header when call gittar api        |
| 🇨🇳 中文    |        pipeline调用gittar增加user-id验证      |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: modules/pipeline/commonutil/thirdparty/gittarutil/commit.go [hunk 1]
```diff
@@ -31,9 +32,10 @@ type Commit struct {
 	CommitMessage string `json:"commitMessage"`
 }
 
-func (r *Repo) GetCommit(ref string) (Commit, error) {
+func (r *Repo) GetCommit(ref, userID string) (Commit, error) {
 	var commit []Commit
 	req := httpclient.New().Get(r.GittarAddr).
+		Header(httputil.UserHeader, userID).
```
Comment (sfwn) [inline modules/pipeline/commonutil/thirdparty/gittarutil/commit.go:38]: internal invoke no need permission.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/precheck/checkers/diceymlchecker/diceyml.go [hunk 2]
```diff
@@ -74,7 +74,7 @@ func checkDiceYmlAndDiceWorkspaceYml(p *spec.Pipeline) error {
 	diceworkspaceymlName = fmt.Sprintf("dice_%s.yml", diceymlworkspace)
 
 	repo := gittarutil.NewRepo(discover.Gittar(), p.CommitDetail.RepoAbbr)
-	diceymlcontent, err := repo.FetchFile(p.GetCommitID(), diceymlName)
+	diceymlcontent, err := repo.FetchFile(p.GetCommitID(), diceymlName, p.GetRunUserID())
```
Comment (sfwn) [inline modules/pipeline/precheck/checkers/diceymlchecker/diceyml.go:77]: precheck will be invoked when pipeline created.
So runUserID always be empty.

Comment (sfwn) [reply modules/pipeline/precheck/checkers/diceymlchecker/diceyml.go:77]: Add a function called GetUserID, get users by runUserID/submitUserID in order.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/precheck/checkers/diceymlchecker/diceyml.go (hunk 2 above)
Comment (littlejiancc) [reply modules/pipeline/precheck/checkers/diceymlchecker/diceyml.go]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**